drivers/vl53l1x/vl53l1x.c

```diff
@@ -0,0 +1,1292 @@
+                              (saved_vhv_timeout & 0x03) + (3 << 2)));
+
+    /* override phasecal */
+    uint8_t reg;
+    EXEC_RET(_write_byte(dev, VL53L1X_PHASECAL_CONFIG__OVERRIDE, 0x01));
+    EXEC_RET(_read_byte (dev, VL53L1X_PHASECAL_RESULT__VCSEL_START, &reg));
+    EXEC_RET(_write_byte(dev, VL53L1X_CAL_CONFIG__VCSEL_START, reg));
+
+    return VL53L1X_OK;
+}
+
+#endif /* MODULE_VL53L1X_BASIC */
+
+static int _update_dss(const vl53l1x_t *dev, _vl53l1x_raw_data_t* raw)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(raw != NULL);
+    DEBUG_DEV("raw=%p", dev, raw);
+
+    uint16_t spad_count;
+    uint32_t peak;
+    uint32_t ambient;
+
+    spad_count  = (uint16_t)raw->dss_actual_effective_spads_sd0_hi << 8;
+    spad_count |= raw->dss_actual_effective_spads_sd0_lo;
+
+    peak  = (uint32_t)raw->peak_signal_count_rate_crosstalk_corrected_mcps_sd0_hi << 8;
+    peak |= raw->peak_signal_count_rate_crosstalk_corrected_mcps_sd0_lo;
+
+    ambient  = (uint32_t)raw->ambient_count_rate_mcps_sd0_hi << 8;
+    ambient |= raw->ambient_count_rate_mcps_sd0_lo;
+
+    if (spad_count != 0) {
+        uint32_t total_rate;
+        /* Calc total rate per spad */
+        total_rate = peak + ambient;
+        /* clip to 16 bits */
+        total_rate = (total_rate > 0xffff) ? 0xffff : total_rate;
+        /* shift up to take advantage of 32 bits */
+        total_rate <<= 16;
+
+        total_rate /= spad_count;
+
+        if (total_rate != 0) {
+            /* get the target rate and shift up by 16 */
+            uint32_t req_spads = ((uint32_t)_target_rate << 16) / total_rate;
+            /* clip to 16 bits */
+            req_spads = (req_spads > 0xffff) ? 0xffff : req_spads;
+
+            /* override DSS config */
+            EXEC_RET(_write_word(dev, VL53L1X_DSS_CONFIG__MANUAL_EFFECTIVE_SPADS_SELECT,
+                                      req_spads));
+            /*
+             * VL53L1X_DSS_CONFIG__ROI_MODE_CONTROL should already be set to
+             * REQUESTED_EFFFECTIVE_SPADS
+             */
+
+             return VL53L1X_OK;
+        }
+    }
+
+    /*
+     * If we reached this point, it means something above would have resulted
+     * in a divide by zero. We want to gracefully set a spad target, not just
+     * exit with an error
+     */
+    /* set target to mid point */
+    EXEC_RET(_write_word(dev, VL53L1X_DSS_CONFIG__MANUAL_EFFECTIVE_SPADS_SELECT,
+                              0x8000));
+    return VL53L1X_OK;
+}
+
+/* Copyrighted POLOLU / ST API code end */
+
+#define VL53L1X_BUFSIZ   32
+static uint8_t _buffer[VL53L1X_BUFSIZ] = {};
+
+static int _read_byte(const vl53l1x_t *dev, uint16_t index, uint8_t *data)
+{
+    return _read (dev, index, data, 1);
+}
+
+static int _read_word(const vl53l1x_t *dev, uint16_t index, uint16_t *data)
+{
+    uint8_t bytes[2];
+    EXEC_RET(_read (dev, index, bytes, 2));
+    *data = (bytes[0] << 8) + (bytes[1]);
```

Comment:
```suggestion
    *data = ((uint16_t)bytes[0] << 8) + (bytes[1]);
```

I *think* this is correct without the cast suggested, as all types smaller than `int` get promoted up to `int` prior to shifting. And `int` is if I recall correctly guaranteed to be at least 16 bits wide. But I would have to consult the C standard to be sure, and at this point I think it would be easier to just add the cast.

Reply:
Fixed with https://github.com/RIOT-OS/RIOT/pull/10363/commits/7c96913c76cd0c89e4b2bcaf66e337e79e7fe63d.